collect_app/src/main/java/org/odk/collect/android/mainmenu/PermissionsDialogFragment.kt

```diff
@@ -0,0 +1,41 @@
+package org.odk.collect.android.mainmenu
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Build
+import android.os.Bundle
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.odk.collect.android.R
+import org.odk.collect.permissions.PermissionListener
+import org.odk.collect.permissions.PermissionsProvider
+
+class PermissionsDialogFragment(
```

Comment:
I was also thinking if they tap outside, it would dismiss the dialog. People may think that tapping "cancel" would mean that they are opting out of these permissions when it's really just an informational dialog. We could have the X at the top so they can dismiss it, but we don't want them to be able to "cancel" anything because that's not actually happening.

Reply:
We could keep this as-is and add analytics for dismissing and OK clicks to see how this does?

Reply:
Yeah sounds good to me.

Reply:
Ok I've added those events in!